server/resp/src/test/java/org/infinispan/JsonCommandsTest.java

```diff
@@ -0,0 +1,319 @@
+            .hasMessageStartingWith("ERR ");
+   }
+
+   @Test
+   public void testJSONSETPaths() {
+      String key = k();
+      JsonPath jp = new JsonPath("$");
+      JsonValue doc = new DefaultJsonParser().createJsonValue("{\"key\":\"value\"}");
+      assertThat(redis.jsonSet(key, jp, doc)).isEqualTo("OK");
+
+      jp = new JsonPath("$.key1");
+      var newNode = new DefaultJsonParser().createJsonValue("\"value1\"");
+      assertThat(redis.jsonSet(key, jp, newNode)).isEqualTo("OK");
+      List<JsonValue> result = redis.jsonGet(key);
+      assertThat(result).hasSize(1);
+      assertThat(compareJSONSet(result.get(0), doc, "$.key1", newNode)).isEqualTo(true);
+
+      doc = result.get(0);
+      jp = new JsonPath("$.key");
+      newNode = new DefaultJsonParser().createJsonValue("{\"key_l1\":\"value_l1\"}");
+      assertThat(redis.jsonSet(key, jp, newNode)).isEqualTo("OK");
+      result = redis.jsonGet(key);
+      assertThat(result).hasSize(1);
+      assertThat(compareJSONSet(result.get(0), doc, "$.key", newNode)).isEqualTo(true);
+
+      jp = new JsonPath("$.lev1");
+      doc = new DefaultJsonParser().createJsonValue("{\"keyl1\":\"valuel1\"}");
+      assertThat(redis.jsonSet(key, jp, doc)).isEqualTo("OK");
+
+      jp = new JsonPath("$.lev1.lev2.lev3");
+      doc = new DefaultJsonParser().createJsonValue("{\"keyl2\":\"valuel2\"}");
+      assertThat(redis.jsonSet(key, jp, doc)).isNull();
+   }
+
+   @Test
+   public void testJSONGET() {
+      JsonPath jp = new JsonPath("$");
+      JsonValue jv = new DefaultJsonParser().createJsonValue("""
+               { "key1":"value1",
+                 "key2":"value2"
+               }
+            """);
+      assertThat(redis.jsonSet(k(), jp, jv)).isEqualTo("OK");
+      JsonPath jp1 = new JsonPath("$.key1");
+      var result = redis.jsonGet(k(), jp1);
+      assertThat(compareJSONGet(result.get(0), jv, jp1)).isEqualTo(true);
+
+      JsonPath jp2 = new JsonPath("$.key2");
+      result = redis.jsonGet(k(), jp1, jp2);
+      assertThat(compareJSONGet(result.get(0), jv, jp1, jp2)).isEqualTo(true);
+
+      JsonPath jp3 = new JsonPath("$.key3");
+      result = redis.jsonGet(k(), jp1, jp2, jp3);
+      assertThat(compareJSONGet(result.get(0), jv, jp1, jp2, jp3)).isEqualTo(true);
+
+      result = redis.jsonGet(k(), jp3);
+      assertThat(compareJSONGet(result.get(0), jv, jp3)).isEqualTo(true);
+
+      // One path multiple results
+      JsonPath jp4 = new JsonPath("$.*");
+      result = redis.jsonGet(k(), jp4);
+      assertThat(compareJSONGet(result.get(0), jv, jp4)).isEqualTo(true);
+
+      // Multiple path multiple results
+      JsonPath jp5 = new JsonPath("$.key1");
+      result = redis.jsonGet(k(), jp4, jp5);
+      assertThat(compareJSONGet(result.get(0), jv, jp4, jp5)).isEqualTo(true);
+   }
+
+   @Test
+   public void testJSONGETPrettyPrinter() {
+      JsonPath jp = new JsonPath("$");
+      JsonValue jv = new DefaultJsonParser().createJsonValue("""
```

Comment:
Should we have a test that has mixed whitespace and possibly not pretty print ?

Also what about commands that use different separators?

Reply:
> Should we have a test that has mixed whitespace and possibly not pretty print ?
> 
added

> Also what about commands that use different separators?

I can't understand, could you provide an example pls?